sklearn/utils/validation.py

```diff
@@ -522,6 +522,7 @@ def _ensure_sparse_format(
         dtype = spmatrix.dtype
 
     changed_format = False
+    spmatrix_type = type(spmatrix).__name__
```

Comment:
Calling this `spmatrix` is a bit confusing because it can be sparse array.

I think we should rename `spmatrix` and `spmatrix_type` to `sparse_data` and `sparse_data_type_name` (or `sparse_container_*`).